pandas/core/arrays/base.py

```diff
@@ -578,6 +578,11 @@ def dropna(self):
         """
         return self[~self.isna()]
 
+    def diff(self, periods: int = 1):
+        if hasattr(self, "__sub__"):
```

Comment:
I'm not thrilled about this, but it may be unavoidable. Happy to hear of an alternative.

Reply:
But if it doesn't support `-`, the operation will also raise a TypeError? What you are doing below yourself?

Reply:
Right. I meant the need for the hasattr just to please the type checker. If you do a diff on an EA that doens't implement `__sub__` you'll get the TypeError either way.